src/server.rs

```diff
@@ -205,6 +206,9 @@ pub async fn run(listener: TcpListener, shutdown: impl Future) -> crate::Result<
     // the `mpsc` channel will close and `recv()` will return `None`.
     let _ = shutdown_complete_rx.recv().await;
 
+    // Signal the 'Db' instance to shutdown the task that purges expired keys
+    db.shutdown_purge_task();
+
     Ok(())
```

Comment:
I was just about to say that since this returns a `Result`, then the method could return without hitting the `shutdown_purge_task` call, however looking more closely it doesn't seem like the method can ever fail. It probably shouldn't be returning an `Ok(())` in the first place.

Anyway, there's still a situation where we might never hit the `shutdown_purge_task` call, namely the situation where the code panics. A way around this is to wrap `db` in another struct, and to call the method from the destructor of that new struct.

Reply:
Makes sense, I removed the `OK(())` from this method. Also added a struct which wraps the `Db`. This way the shutdown is more nicely encapsulated in the db module. I'm not completely happy with the name `DbHolder` though.

Reply:
How about the name `DbDropGuard`?

Reply:
Yes I like that, will change it